[Weiennn] iP

Weiennn opened the pull request:
#Botty
> "It is what it is" -me

Botty is your all-in-one companion for remembering things you need to do. It's 
- CLI based
- Easy to learn 
- ~~Fast~~ _SUPER_ FAST to use

All you need to do is,
1. Download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html#3-decide-on-an-overall-project-direction-user-profile-problem-addressed-before-the-tutorial)
2. Double-click it
3. Add you tasks
4. Let it manage your tasks for you!😉

And it is **ABSOLUTELY FREE**

Features
- [x] Managing tasks
- [ ] Managing deadlines (Coming soon)
- [ ] Reminders (Coming soon)

---------------------------------
If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Weiennn committed:
Rename, Greet, Exit

Weiennn committed:
Level-1 Echo

Weiennn committed:
Level 2. Add, List

Weiennn committed:
Level 3. Mark as Done

Weiennn committed:
Level 4. ToDos, Events, Deadlines

Weiennn committed:
Level 5. Handle Errors

Weiennn committed:
Level 6. Delete

Weiennn committed:
added javadoc comments

Weiennn committed:
implement more OOP

Weiennn committed:
Extracted UI, Storage, Parser, TaskList classes

Weiennn committed:
Level-7. Save

Weiennn committed:
Level-8. Dates and Times

Weiennn committed:
storage changes

Weiennn committed:
Merge Level 8 change

Weiennn committed:
storage.txt changes

Weiennn committed:
Merging Level 7 changes

Weiennn committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Weiennn committed:
A-JUnit

Weiennn committed:
Level-9

Weiennn committed:
A-CodingStandard

Weiennn committed:
Add Javadocs

Weiennn committed:
Merge branch 'branch-A-CodingStandard'

Weiennn committed:
merge conflicts

Weiennn committed:
merge conflicts

Weiennn committed:
Level-10

coderhuang559 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+     * @param by          The due date/time for the deadline task.
+     * @param bool        The completion status for the deadline task.
+     */
+    public Deadline(String description, String by, Boolean bool) {
```
Ambiguous name. Perhaps a more boolean-sounding name like isDone could be used instead?

Weiennn replied:
yepp good catch, I will change that

coderhuang559 commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,21 @@
+import main.java.ui.Ui;
+
+public class DeleteCommand extends Command {
+    private int deleteTaskIndex;
```
deleteTaskIndex sounds more like a method name. Maybe something like indexToDelete?

Weiennn replied:
yepp it does sound like a method name

coderhuang559 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,60 @@
+        case "find":
+            return new FindCommand(stringArray[1]);
+        default:
+            throw new UnknownCommandException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n");
```
clean code! very nice.

coderhuang559 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Represents a todo task, which is a task without any date/time attached to it.
+ */
+public class Todo extends Task {
```
Class names are in Pascal Case i.e. ToDo.

coderhuang559 commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,27 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class DeadlineTest {
+    @Test
+    public void deadline_markTest_success() {
```
Good job following test method naming conventions!

coderhuang559 commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,27 @@
+        t.markUndone();
+        assertEquals(" ", t.getStatusIcon());
+    }
+
```
Maybe you can try to test these methods more extensively, like marking first task, last task, task out of index etc.

coderhuang559 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,13 @@
+ * The `Command` class is an abstract class for different command implementations.
+ * Subclasses of this class represent specific actions that can be executed on a task list.
+ */
+public abstract class Command {
```
Good job implementing Command class even though it's not required!!

coderhuang559 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    public String getSaveDescription() {
+        return (this.isDone
+                ? "| 1 | " + this.description
```
"|" can be left out of the logic operation for clearer compartmentalisation

coderhuang559 reviewed (commented):
Overall very clean code. Good job!

Weiennn committed:
commit branch

Weiennn committed:
to push branch

Weiennn committed:
Merge branch 'branch-Level-10'

chewjh1234 commented on src/main/java/Botty.java:
```diff
@@ -0,0 +1,69 @@
+        this.storage = new Storage("./storage.txt");
+        this.storage.loadTaskList(this.taskList);
+    }
+    public static void main(String[] args) {
```
Perhaps there could be a blank line before the main method to improve code readability.

chewjh1234 commented on src/test/java/TodoTest.java:
```diff
@@ -0,0 +1,27 @@
+import main.java.Task;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class TodoTest {
```
Similar to what @coderhuang559 has mentioned for your `Todo` class, perhaps your `TodoTest` class would also have to be in Pascal Case.

chewjh1234 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,97 @@
+            while (sc.hasNextLine()) {
+                String input = sc.nextLine();
+                String[] task = input.split(" \\| ", 5);
+                Boolean taskBool = task[1] == "1";
```
Perhaps `isDone` would be more appropriate here too.

chewjh1234 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,60 @@
+package main.java.parser;
+
+import main.java.command.*;
```
Perhaps you could look into whether this wildcard import is necessary. While I understand that you will need all commands, CS2103T coding standards seem to discourage the use of wildcard imports.

kristayeo replied:
should specify exactly what you're importing, shouldnt use the *

chewjh1234 reviewed (commented):
Your code looks good! Only minor changes, as pointed out by the previous reviewer, are needed. I like how there are many different commands and exceptions, and they are evident that you have achieved the 'stretch goals' for iP. Well done!

kristayeo commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,39 @@
+     */
+    public Todo(String description, Boolean bool) {
+        super(description, bool);
+    }
```
could add a line here for some spacing between the methods

kristayeo commented on src/main/java/command/MarkCommand.java:
```diff
@@ -0,0 +1,23 @@
+package main.java.command;
```
LGTM! OOP is clear and the classes are separated neatly, code is easy to digest

Weiennn committed:
Add assertions to

Weiennn committed:
Add Assertion to make code more reliable

There is no check for empty Fxml Loader in main. There is no check for empty response in mainwindow.

To ensure code is more reliable

Weiennn committed:
Remove Parser class wildcard import

Parser class has a command wildcard import

Wildcard import is strongly discouraged

Weiennn committed:
Merge pull request #1 from Weiennn/branch-A-CodeQuality

Remove Parser class wildcard import

Weiennn committed:
Merge branch 'branch-A-Assertions'

Weiennn committed:
BCD-Extension

Weiennn committed:
Merge pull request #2 from Weiennn/branch-C-Undo

BCD-Extension

Weiennn committed:
Change font for GUI

Weiennn committed:
UG

Weiennn committed:
Add UG

Weiennn committed:
Add javadoc comments

Weiennn committed:
Change Ui.png

Weiennn committed:
Edit UG
